matlab/MeCorr/MATLAB_CODE_allrxns.m

```diff
@@ -0,0 +1,419 @@
+            %             [fluxstate_gurobi] = constrain_flux_regulation(model2,onreactions,offreactions,kappa,rho,epsilon,MODE,[],minfluxflag);
+            %             grate_ccle_exp_soft(i,3) = fluxstate_gurobi(rxnpos); %methylation flux
+            %         end
+            
+            for rxncount = 1:length(model2.rxns)
+                if basalflag
+                    %minfluxflag = 1;
+                    %[~, grate, solverobj_ccle] = constrain_flux_regulation(model2,onreactions,offreactions,kappa,rho,epsilon,MODE,[], minfluxflag); % impact on growth
+                    model2.c(rxncount) = epsilon_methylation ;
+                    [fluxstate_gurobi] = constrain_flux_regulation(model2,onreactions,offreactions,kappa,rho,epsilon,MODE,[],minfluxflag);
+                    g_rate(i,rxncount) = fluxstate_gurobi(rxncount); %methylation flux
+                end
+            end
+        else
+            grate_ccle_exp_soft(i,:) = NaN;
+        end
+    end
+    % Save in local github repo and another local folder. wd is MeCorr
+    s_var= num2str(name(i_weight));
+    s_name= strcat('grate_', s_var, 'FBA_allrxns');
+    s_path1= strcat('VariablesSaved/MaxAllRxns_Results/', s_name);
+    save(s_path1, 'g_rate');
+    s_path2= strcat('./../../../MaxAllRxns_local/', s_name);
+    save(s_path2, 'g_rate');
+    
+    % figure; h = histogram(grate_ccle_exp_soft(:,3),70);
+    %  xlabel('Predicted methylation flux')
+    %  ylabel('Total cell lines')
+    % title('Distribution of methylation flux among CCLE cell lines','fontweight','normal')
+    
+    %fluxesAll_1pFBA= fluxes_allrxns;
+    % save('VariablesSaved\grate_3FBA_allRxns', 'g_rate');
+    % save('VariablesSaved\fluxesAll_1FBA', 'fluxes_allrxns');
+    % save('VariablesSaved\grate_1E-6', 'grate_ccle_exp_soft');
+    % save('VariablesSaved\fluxstate_1E-6', 'fluxstate_gurobi');
+end
+%% Correlation between flux and auc for a reaction
+% flux_allrexns: 1031 cell lines by 3777 reactions
+% drug_auc_expt: extract auc values for the 8 methyl drugs
+% 8 x 3000 rhos
+%fluxes_allrxns= g_rate;
+rho= NaN(height(hmei_list),length(model2.rxns));
+rho_p= NaN(height(hmei_list),length(model2.rxns));
+for j = 1:height(hmei_list)
+    fx = find(ismember(ctd2compoundidname_name, hmei_list(j,1)));
+    ix = ismember(drug_auc_expt(:,3), ctd2compoundidname_id(fx,1)); sum(ix);
+    hmei_auc_dat = drug_auc_expt(ix,:);
+    
+    % Only uses the experiments of fluxes_allrxns that also are in hmei_auc_dat
+    [ix, pos] = ismember(hmei_auc_dat(:,1), exptidcelllinemediamatch(:,1)); sum(ix); 
+    v1 = hmei_auc_dat(:,2); v1= table2array(v1);
+    for nCol=1:size(fluxes_allrxns,2)
+        v4= fluxes_allrxns(pos,nCol);
+        [rho(j,nCol), rho_p(j,nCol)]= corr(v4, v1, 'rows', 'complete');
+    end
+end
+disp('rho & p-value calculated')
+%% Create Table of Significant Reactions by Correlation value
+% Workflow: Change threshold. Change struct field name (e.g. above3)
+% accordingly
+sigTF= (abs(rho) > 0.3);

```

Comment:
Why are you taking the absolute value of rho? Negative values are important and have a statistical interpretation.

Reply:
Also if you want to get significant reactions, you would want to filter using the p-values as well as R value. That is, you want the reaction to have an R-value >= 0.3 and a p-value < 0.05.

Reply:
Line 195: I coded abs(rho) > 0.3 so that the negative correlations below -0.3 would be selected as well. I am including those correlations in downstream analysis.

Reply:
You should create two variables have rho > 0.3 and rho < 0.3, unless you wrote in a way to distinguish between the negatively and positively correlated entries by saving their respective indices first.